backend/app/models/user.py

```diff
@@ -10,10 +10,12 @@ class User(db.Model):
     id = db.Column(db.Integer, primary_key=True, nullable=False)
     first_name = db.Column(db.String, nullable=False)
     last_name = db.Column(db.String, nullable=False)
-    auth_id = db.Column(db.String, nullable=False)
+    auth_id = db.Column(db.String, nullable=True)
```

Comment:
we should only allow auth_id to be nullable when the user's status is "Invited". I asked chatgpt (bestie) how to do this, and it looks like this needs to be added:
```
    __table_args__ = (
        CheckConstraint(
            "(user_status = 'INVITED' AND auth_id IS NULL) OR (user_status != 'INVITED' AND auth_id IS NOT NULL)",
            name="check_auth_id_nullable"
        ),
    )
```

Reply:
you can probably add it around line 19